changed/fixed references

ThomasBreuer opened the pull request:
- turned [GDF20] from `@article` into `@misc` and set `url`;
  this way, the erroneous double comma disappears and we get a link to the paper

- (this is perhaps controversial)
  replaced the entries by the ones from MathSciNet where possible;
  this yields links for [EHV92], [KL91], [Wil13],
  and more potentially useful information
  (for example, `mrnumber` could be used to produce links to the
  corresponding review in MathSciNet; the analogous processor in GAPDoc
  does this automatically, and I find this very useful)

  The data of the old and new entries are not completely consistent:
  [GP07] is assigned to the year 2008 in MathSciNet, and
  [PSS11] has a different `journal`.

(Documenter.jl's conversion from BibTeX format to HTML has some shortcomings:
Capitalization syntax (like `{G}ram-{S}chmidt`) is not supported,
the `NOTE` field is not honoured, `--` is not converted, etc.,
and finally the HTML is not valid.)

ThomasBreuer committed:
changed/fixed references

- turned [GDF20] from `@article` into `@misc` and set `url`;
  this way, the erroneous double comma disappears and we get a link to the paper

- (this is perhaps controversial)
  replaced the entries by the ones from MathSciNet where possible;
  this yields links for [EHV92], [KL91], [Wil13],
  and more potentially useful information
  (for example, `mrnumber` could be used to produce links to the
  corresponding review in MathSciNet; the analogous processor in GAPDoc
  does this automatically, and I find this very useful)

  The data of the old and new entries are not completely consistent:
  [GP07] is assigned to the year 2008 in MathSciNet, and
  [PSS11] has a different `journal`.

(Documenter.jl's conversion from BibTeX format to HTML has some shortcomings:
Capitalization syntax (like `{G}ram-{S}chmidt`) is not supported,
the `NOTE` field is not honoured, `--` is not converted,
and finally the HTML is not valid.)

fingolfin commented on docs/oscar_references.bib:
```diff
@@ -1,7 +1,8 @@
-@article{GDF20,
+@misc{GDF20,
```
I would also do this (to match the bibkey the BibTeX style `alpha` would produce):
```suggestion
@misc{DF20,
```

ThomasBreuer replied:
Yes.
If we want our keys to be consistent with the `alpha` style then four more keys have to be adjusted.
I have updated the pull request accordingly.

fingolfin commented:
DocumenterCitation has been massively improved in the past days (I helped a bit with it) -- the HTML should be correct now (or at least better) and `--` converted. Curly braces may also be better supported soon (but this has more ramifications than just capitalizations). Regarding "note", I am sure it could be supported, just need to decide how

fingolfin reviewed (approved):
Looks good to me

wdecker commented:
I am fine with replacing bibtex entries by those from MathSciNet. @ThomasBreuer: Thanks for doing this!

ThomasBreuer committed:
adjusted bib keys to the `alpha` style

as suggested by @fingolin

(The key `[OMdCS00]`, instead of `[OS]`, is a bit surprising.)
